From 8e724faac114e73c2a32d5ce7556c0f1fa37a658 Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Mon, 9 Apr 2018 17:58:44 +0800
Subject: [PATCH 3586/5242] MLK-17991-6 gpu: imx: imx8_dprc: Do not support
 YUV pfmts for WARP SC resources

commit  9c796d8a87e780b24dfe04cff12f7e056e0c0cbf from
https://source.codeaurora.org/external/imx/linux-imx.git

The dpu common driver is not supporting the YUV pixel formats for
the fetchwarp fetch units yet, so let's explicitly tell the callers
of dprc_format_supported() that DPR currently doesn't support
YUV pixel formats for WARP SC resources.

Signed-off-by: Liu Ying <victor.liu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/imx8_dprc.c |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/gpu/imx/imx8_dprc.c b/drivers/gpu/imx/imx8_dprc.c
index 6a56ad7..5bee77d 100644
--- a/drivers/gpu/imx/imx8_dprc.c
+++ b/drivers/gpu/imx/imx8_dprc.c
@@ -594,6 +594,8 @@ bool dprc_format_supported(struct dprc *dprc, u32 format, u64 modifier)
 		switch (dprc->sc_resource) {
 		case SC_R_DC_0_FRAC0:
 		case SC_R_DC_1_FRAC0:
+		case SC_R_DC_0_WARP:
+		case SC_R_DC_1_WARP:
 			return false;
 		}
 		return modifier == DRM_FORMAT_MOD_NONE;
@@ -602,6 +604,8 @@ bool dprc_format_supported(struct dprc *dprc, u32 format, u64 modifier)
 		switch (dprc->sc_resource) {
 		case SC_R_DC_0_FRAC0:
 		case SC_R_DC_1_FRAC0:
+		case SC_R_DC_0_WARP:
+		case SC_R_DC_1_WARP:
 			return false;
 		}
 		return (dprc->has_aux_prg &&
-- 
1.7.9.5

